ssc/libssc.c

```diff
@@ -254,210 +551,79 @@ static PetscErrorCode PCPatchCreateCellPatches(PC pc)
      * actually remember the cells. */
     for ( PetscInt v = vStart; v < vEnd; v++ ) {
         PetscInt ndof, off;
+        PetscHashIIter hi;
+
         ierr = PetscSectionGetDof(cellCounts, v, &ndof); CHKERRQ(ierr);
         ierr = PetscSectionGetOffset(cellCounts, v, &off); CHKERRQ(ierr);
         if ( ndof <= 0 ) {
             continue;
         }
-        ierr = DMPlexGetTransitiveClosure(dm, v, PETSC_FALSE, &closureSize, &closure); CHKERRQ(ierr);
+        ierr = patch->patchconstructop((void*)patch, dm, v, ht); CHKERRQ(ierr);
+        ierr = PCPatchCompleteCellPatch(dm, ht, cht);
         ndof = 0;
-        for ( PetscInt ci = 0; ci < closureSize; ci++ ) {
-            const PetscInt c = closure[2*ci];
-            if (cStart <= c && c < cEnd) {
-                cellsArray[ndof + off] = c;
+        PetscHashIIterBegin(cht, hi);
```

Comment:
Again, segfault if `cht` is empty.

Reply:
Now safe, the loop continues in the empty case on line 569.